src/components/text-input/text-input.hbs

```diff
@@ -1,33 +1,69 @@
-<div class="bx--form-item">
+<div data-text-input class="bx--form-item">
   <label for="text-input-3" class="bx--label">Text Input label</label>
-  <input id="text-input-3" type="text" class="bx--text-input{{#if light}} bx--text-input--light{{/if}}" placeholder="Optional placeholder text">
+  <input id="text-input-3" {{#if password}} type="password" {{else}} type="text" {{/if}} class="bx--text-input{{#if light}} bx--text-input--light{{/if}}"
+    {{#if password}} data-toggle-password-visibility{{/if}} placeholder="Optional placeholder text"> {{#if password}}
+  <button class="bx--text-input--password__visibility">
+    <svg width="16" height="16" viewBox="0 0 16 16">
+      <path d="M11.846 3.45L15.293.007 16 .714l-3.284 3.281c1.261.902 2.377 2.212 3.347 3.93C14.02 11.642 11.333 13.5 8 13.5c-1.392 0-2.667-.324-3.822-.973L.703 16l-.706-.708 3.323-3.32C2.071 11.042.976 9.694.035 7.924 2.012 4.308 4.667 2.5 8 2.5c1.395 0 2.677.317 3.846.95zm-6.928 8.338c.944.477 1.97.712 3.082.712 2.795 0 5.076-1.483 6.907-4.568-.866-1.417-1.833-2.486-2.91-3.219l-1.55 1.55a3 3 0 0 1-4.185 4.182l-1.344 1.343zm-.882-.533l1.518-1.517A3 3 0 0 1 9.74 5.556l1.364-1.363A7.02 7.02 0 0 0 8 3.5c-2.798 0-5.047 1.439-6.819 4.432.842 1.465 1.792 2.568 2.855 3.323zm2.948-1.532a2 2 0 0 0 2.74-2.738l-2.74 2.738zm-.707-.707l2.74-2.738a2 2 0 0 0-2.74 2.738z"></path>
+    </svg>
+  </button>
+  {{/if}}
 </div>
 
-<div class="bx--form-item">
+<div data-text-input class="bx--form-item">
```

Comment:
Nitpick: Do we want to prevent `TextInput` JS class from being instantiated for non-password text inputs? 🤔

Reply:
I added the `[data-text-input]` on `div.bx--form-item` in order to differentiate text inputs from other components that are also wrapped with `div.bx--form-item`. Some of those components like `Checkbox` and `ComboBox` also have identifying classes on that div like `.bx--combo-box` or `.bx--checkbox-wrapper`, but other components like `DatePicker` or `TextInput` do not.

My thinking was that it might be convenient for users that want to select the TextInput `div.bx--form-item`, but in retrospect it might be better to use something like `.bx--text-input-wrapper` rather than a data attribute. And if we do add a wrapper class, then I think we should prevent `TextInput` JS class from being instantiated for non-password text inputs. So a password text input would be `div.bx--form-item[data-text-input]` while a non-password text input would be `div.bx--form-item.bx--text-input-wrapper`. What do you think @asudoh?

Reply:
@emyarod Thank you for putting your thoughts on this! I think it makes sense.